File: app/src/main/java/org/wikipedia/feed/random/RandomCardView.java
package org.wikipedia.feed.random;

import android.content.Context;

import androidx.annotation.NonNull;

import org.wikipedia.feed.featured.FeaturedArticleCardView;
import org.wikipedia.feed.view.CardFooterView;

public class RandomCardView extends FeaturedArticleCardView {
    public interface Callback {
        void onRandomClick(@NonNull RandomCardView view);
    }

    public RandomCardView(Context context) {
        super(context);
    }

    @Override
    public CardFooterView.Callback getFooterCallback() {
        return () -> {
            if (getCallback() != null && getCard() != null) {
                getCallback().onRandomClick(RandomCardView.this);
            }
        };
    }
}

File: app/src/main/java/org/wikipedia/feed/random/RandomCardView.kt
package org.wikipedia.feed.random

import android.content.Context
import org.wikipedia.feed.featured.FeaturedArticleCardView
import org.wikipedia.feed.view.CardFooterView

class RandomCardView(context: Context) : FeaturedArticleCardView(context) {
    interface Callback {
        fun onRandomClick(view: RandomCardView)
    }

    override val footerCallback: CardFooterView.Callback
        get() = CardFooterView.Callback {
            card?.let {
                callback?.onRandomClick(this@RandomCardView)
            }
        }
}
